examples/mqtt-paho/main.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2019
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       MQTT-paho example
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <thread.h>
+#include <mqtt.h>
+#include <log.h>
+#include "msg.h"
+#include "lwip.h"
+#include "lwip/netif.h"
+#include "net/ipv6/addr.h"
+#include "net/ipv4/addr.h"
+#include "MQTTClient.h"
+#include "shell.h"
+
+#define REMOTE_IP_ADDRESS        "fe80::18cf:f9ff:fe1c:afd2"
+#define MAX_MSGS                12
+
+static char* topic = "test_from_riot";
+MQTTClient client;
+static char _sender_stack[THREAD_STACKSIZE_DEFAULT];
+
+static void _messageArrived(MessageData* data)
+{
+    printf("mqtt_example: message arrived on topic"
+          " %.*s: %.*s\n\n", (int)data->topicName->lenstring.len, data->topicName->lenstring.data,
+      (int)data->message->payloadlen, (char *)data->message->payload);
+}
+
+static int ifconfig(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+    for (struct netif *iface = netif_list; iface != NULL; iface = iface->next) {
+        printf("%s_%02u: ", iface->name, iface->num);
+#ifdef MODULE_LWIP_IPV6
+        char addrstr[IPV6_ADDR_MAX_STR_LEN];
+        for (int i = 0; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
```

Comment:
This is the only line in this file that explicitly requires something from LWIP (`LWIP_IPV6_NUM_ADDRESSES`). Is there a generic function that could be used instead of this ?